Modules/hpe3par_remote_copy.py

```diff
@@ -0,0 +1,1442 @@
+    hpe3par_remote_copy:
+      storage_system_ip: 10.10.10.1
+      storage_system_password: password
+      storage_system_username: username
+      state: startrcopy
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from hpe3par_sdk import client
+except ImportError:
+    client = None
+
+def create_remote_copy_group(
+            client_obj,
+            storage_system_username,
+            storage_system_password,
+            remote_copy_group_name,
+            domain,
+            remote_copy_targets,
+            local_user_cpg,
+            local_snap_cpg
+            ):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "Remote Copy Group create failed. Storage system username or password is null",
+            {})
+    if remote_copy_group_name is None:
+        return (False, False, "Remote Copy Group create failed. Remote Copy Group name is null", {})
+    if len(remote_copy_group_name) < 1 or len(remote_copy_group_name) > 31:
+        return (False, False, "Remote Copy Group create failed. Remote Copy Group name must be atleast 1 character and not more than 31 characters", {})
+    if (not local_user_cpg and local_snap_cpg) or (local_user_cpg and not local_snap_cpg):
+        return (False, False, "Either both local_user_cpg and local_snap_cpg must be present, or none of them must be present", {})
+    targets_transformed = []
+    target_names_list = []
+    for dict1 in remote_copy_targets:
+        target={}
+        for key in dict1.keys():
+            if key == 'target_name':
+                if dict1[key] is None:
+                    return (False, False, "Remote Copy Group create failed.Target name is null", {})
+                else:
+                    target['targetName']=dict1.get(key)
+                    target_names_list.append(dict1.get(key))
+            elif key == 'target_mode':
+                if dict1.get(key) == 'sync':
+                    target['mode'] = 1
+                elif dict1.get(key) == 'periodic':
+                    target['mode'] = 3
+                elif dict1.get(key) == 'async':
+                    target['mode'] = 4
+                else:
+                    return (False, False, "Remote Copy Group create failed.Target mode is invalid", {}) 
+            elif key == 'user_cpg':
+                target['userCPG']=dict1.get(key)
+            elif key == 'snap_cpg':
+                target['snapCPG']=dict1.get(key)
+            else:
+                return (False, False, "Remote Copy Group create failed. Wrong parameter name %s is given." % key, {}) 
+        targets_transformed.append(target)
+    remote_copy_targets = targets_transformed
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        target_name = client_obj.getStorageSystemInfo()['name']
+        if target_name in target_names_list:
```

Comment:
I believe we should not fail in this case. The other targets can still be added to the group, right? Fail only in case there exists only one target which is same as source